openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/visits/PatientDashboardVisitsPresenter.java

```diff
@@ -28,29 +28,37 @@
 
 public class PatientDashboardVisitsPresenter extends PatientDashboardMainPresenterImpl implements PatientDashboardContract.PatientVisitsPresenter {
 
+    private final String patientId;
     private PatientDashboardContract.ViewPatientVisits mPatientVisitsView;
     private VisitDAO visitDAO;
+    private PatientDAO patientDAO;
 
-    public PatientDashboardVisitsPresenter(String id, PatientDashboardContract.ViewPatientVisits mPatientVisitsView) {
-        this.mPatient = new PatientDAO().findPatientByID(id);
+    public PatientDashboardVisitsPresenter(String patientId, PatientDashboardContract.ViewPatientVisits mPatientVisitsView) {
+        this.patientId = patientId;
         this.mPatientVisitsView = mPatientVisitsView;
         this.mPatientVisitsView.setPresenter(this);
         this.visitDAO = new VisitDAO();
+        this.patientDAO = new PatientDAO();
     }
 
     @Override
     public void start() {
-        visitDAO.getVisitsByPatientID(mPatient.getId())
+        patientDAO.findPatientByID(patientId)
```

Comment:
is this necessary at all ? I mean there is patientId already right and we get visits by patientId